### PR TITLE
add cookie prefs management helper

### PR DESCRIPTION
**Pull request by doramatadora**
🐿 v2.8.0

### CODE REVIEW

File: src/js/helpers.ts [hunk 1]
```diff
@@ -184,3 +170,93 @@ export function buildConsentRecord(
...
+
+	document.cookie = cookie.join('; '); // RFC says space after semicolon
+}
+
+interface UnmarshalledCookie {
+	[x: string]: boolean;
+}
+
+function parseConsentCookie(name: string): UnmarshalledCookie | null {
+	const rx = new RegExp(`${name}=(.*?);`);
```
Comment (debugwand) [inline src/js/helpers.ts]: would https://github.com/Financial-Times/n-ui-foundations#cookiestore help?

File: test/unit/helpers.spec.ts [hunk 2]
```diff
@@ -135,4 +137,82 @@ describe('helper functions', () => {
 			expectedResult
 		);
 	});
+
+	describe.only('cookie handling', () => {
```
Comment (debugwand) [inline test/unit/helpers.spec.ts]: remove .only

File: src/js/helpers.ts (hunk 1 above)
Comment (doramatadora) [reply src/js/helpers.ts]: Yes, but in the interest of keeping client-side bundles light, I implemented cookie handling this way. Happy to take your lead on this.

Comment (debugwand) [reply src/js/helpers.ts]: if we can't assume the main app has n-ui-foundations then maybe yes it is better not to pull in something that big

**Pull request merged**